feat: cp-7.61.0 add PNA25 messaging for updated metametrics

ameliejyc opened the pull request:
## **Description**

This PR implements PNA25 (Privacy Notice Announcement) messaging to inform existing users about MetaMetrics updates. The changes introduce a new toast that alerts existing users who have opted into MetaMetrics about the association of in-app activity with on-chain data.

**What is the reason for the change?**
As part of the PNA25 initiative, we need to notify existing users who have opted into MetaMetrics about the updated data collection practices, specifically that some in-app activity will be associated with on-chain data to better understand usage patterns.

**What is the improvement/solution?**
- Added a new Redux action and state management for tracking PNA25 acknowledgement
- Created selectors to determine when the PNA25 toast should be displayed based on user state and feature flags
- Implemented a toast notification on the Wallet screen that appears for existing users with MetaMetrics enabled
- Updated onboarding flow to automatically acknowledge PNA25 for new users
- Refactored legal notices actions from reducer to dedicated actions file
- Updated MetaMetrics description strings to reflect the new data collection approach
- Updated MetaMetrics support URL to the latest documentation

There are very specific testing instructions below as there are several different cases for this.

## **Changelog**

CHANGELOG entry: Added PNA25 toast message to inform existing users about MetaMetrics updates and on-chain data association

## **Related issues**

https://consensyssoftware.atlassian.net/browse/CEUX-728
https://consensyssoftware.atlassian.net/browse/CEUX-729
https://consensyssoftware.atlassian.net/browse/CEUX-727

## **Manual testing steps**

Test changes using the LaunchDarkly feature flag `extension-ux-pna25`

```
Update from previous versions (existing users)

Scenario: MetaMetrics opt in = true

Step 1: Download & install 7.60.0 or earlier
Step 2: Complete onboarding with MetaMetrics opt in = true
Step 3: Update to 7.61
Step 4: Open home screen

Expected Result
Toast appears

 

Scenario: MetaMetrics opt in = false

Step 1: Download & install 7.60.0 or earlier
Step 2: Complete onboarding with MetaMetrics opt in = false
Step 3: Update to  7.61
Step 4: Open home screen

Expected Result
No toast appears

 

Scenario: Feature flag disabled

Step 1: Download & install 7.60.0 or earlier
Step 2: Complete onboarding with MetaMetrics opt in = false
Step 3: Update to  7.61
Step 4: Feature flag is disabled (disable main-dev for flag extension-ux-pna25 (same on Mobile and Extension)
Step 5: Open home screen

Expected Result
No toast appears

 

New install (new users)

Scenario: MetaMetrics opt-in = true

Step 1: Download RC
Step 2: Complete onboarding with MetaMetrics opt in = true
Step 5: Open home screen

Expected Result
No toast appears

 

Scenario: MetaMetrics opt-in = false

Step 1: Download RC
Step 2: Complete onboarding with MetaMetrics opt in = false
Step 5: Open home screen

Expected Result
No toast appears
 

Scenario: Social login onboarding

Step 1: Download RC
Step 2: Complete onboarding with Social login
Step 5: Open home screen

Expected Result
No toast appears


MetaMetrics opt out / opt in

Scenario: Enable MetaMetrics post onboarding

Step 1: Download RC OR follow the steps for existing user
Step 2: Complete onboarding with MetaMetrics opt out
Step 3: Enable MetaMetrics in Settings

Expected Result
No toast appears
```

## **Screenshots/Recordings**

### **Before**


### **After**

During onboarding, the 'Gather basic usage data' copy is updated when the feature flag is on:

<img width="397" height="868" alt="image" src="https://github.com/user-attachments/assets/56f166fa-3b8b-4dc0-8e2b-74044ce7042e" />

When the toast should appear, it looks like this and should persist until either the Learn more or Close button is pressed:

(NOTE: the close button is going to have UI tweaks in a later PR)

<img width="395" height="821" alt="image" src="https://github.com/user-attachments/assets/1df6db8f-3fc8-466a-9529-159031c7fbd0" />

In security settings the Metametrics copy is updated (regardless of feature flag on or off):

<img width="364" height="126" alt="image" src="https://github.com/user-attachments/assets/ec9b116f-626c-4683-a1fb-9fc94689fa78" />



## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.




<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Introduces PNA25 feature-flagged messaging: new Wallet toast, onboarding/Settings acknowledgment flow, state/selectors, and updated copy/URLs for MetaMetrics.
> 
> - **UX / Messaging**
>   - **Wallet**: Displays persistent PNA25 toast with Learn more and Close; tracks impressions/close; uses `selectShouldShowPna25Toast`.
>   - **Onboarding**: Auto-acknowledges PNA25 for new users when flag enabled.
>   - **Opt-in Screen**: Conditioned description shows updated copy when `extensionUxPna25` flag is on.
>   - **Settings**: When enabling MetaMetrics, dispatches PNA25 acknowledgment if needed.
> - **State & Selectors**
>   - New actions: `storePna25Acknowledged`, `storePrivacyPolicyShownDate`, `storePrivacyPolicyClickedOrClosed` moved to `actions/legalNotices`.
>   - Reducer: adds `isPna25Acknowledged`; keeps privacy toast state; exports `newPrivacyPolicyDate`.
>   - Selectors: `selectIsPna25FlagEnabled` (env override `MM_EXTENSION_UX_PNA25`), `selectIsPna25Acknowledged`, `selectShouldShowPna25Toast`, and refactor of privacy toast selector.
> - **Analytics**
>   - Adds `MetaMetricsEvents.TOAST_DISPLAYED` and event tracking for PNA25 toast.
> - **Content & Links**
>   - Updates strings for MetaMetrics descriptions and adds `privacy_policy.pna25_toast_message`.
>   - Updates `HOWTO_MANAGE_METAMETRICS` URL.
> - **Config**
>   - Adds env flag `MM_EXTENSION_UX_PNA25`.
>   - Bumps build/version numbers in CI and iOS project.
> - **Tests**
>   - Adds/updates unit tests for actions, reducer, selectors, OptinMetrics, Settings section, Wallet behavior, Sentry masks, fixtures, and snapshots.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a0ac8da4859c8590aba4785e62d23cc1ee2a44fa. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

ameliejyc committed:
chore: update metametrics Settings string

ameliejyc committed:
chore: update metametrics support url

ameliejyc committed:
feat: add pna25 messaging

ameliejyc committed:
chore: add flag selector test

ameliejyc committed:
chore: add pna25 logic on Metametrics Settings toggle

ameliejyc committed:
chore: refactor and add tests

ameliejyc committed:
chore: run format

ameliejyc committed:
chore: add space after string

NidhiKJha reviewed (commented):
LGTM! Just left one comment where functionality is different from extension

ameliejyc committed:
chore: don't close toast on Learn more click

ameliejyc committed:
chore: remove requirement to disable toast for social login users

ameliejyc committed:
chore: move onboarding pna25 acknowledgement to capture social login onboarders using existing logins

ameliejyc committed:
chore: add MM_EXTENSION_UX_PNA25 env override in case remote flag is unavailable during onboarding

ameliejyc committed:
Merge branch 'main' into ceux-730-pna25-messaging

ameliejyc committed:
chore: add ref to keep toast stable rather than omitting deps

ameliejyc committed:
Merge branch 'main' into ceux-730-pna25-messaging

christopherferreira9 reviewed (dismissed):
Looks good for QA changes!

ameliejyc committed:
chore: add PNA25 toast metrics (#23369)

## **Description**

Adds the ToastDisplayed metric when a) the user sees the toast and b)
when the user closes the toast.

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/CEUX-725

## **Manual testing steps**

Run locally and see the following events on view and close:

```
TRACK event saved {"event": "Toast Displayed", "properties": {"closed": false, "toast_name": "pna25"}, "type": "track"}
```

```
TRACK event saved {"event": "Toast Displayed", "properties": {"closed": true, "toast_name": "pna25"}, "type": "track"}
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Tracks PNA25 toast impressions and closes via new Toast Displayed
analytics event.
> 
> - **Analytics**:
> - Add `TOAST_DISPLAYED` to `EVENT_NAME` and export
`MetaMetricsEvents.TOAST_DISPLAYED`.
> - **Wallet (PNA25 toast)**:
> - Track toast shown with `MetaMetricsEvents.TOAST_DISPLAYED` and
properties `{ toast_name: 'pna25', closed: false }`.
> - Track toast closed in close handler with `{ toast_name: 'pna25',
closed: true }`.
> - Update effect dependencies to include `trackEvent` and
`createEventBuilder`.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
1e2d6c105860e6f4cf98483524936b5f5df91b1b. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

metamaskbot committed:
[skip ci] Bump version number to 3168

ameliejyc committed:
Merge branch 'main' into ceux-730-pna25-messaging

ameliejyc committed:
chore: update Onboarding file with PNA25 changes

himanshuchawla009 reviewed (dismissed):
lgtm

ameliejyc committed:
fix: move PNA25 check to separate effect to catch feature flag update

ameliejyc committed:
Merge branch 'main' into ceux-730-pna25-messaging

NicolasMassart commented on app/components/Views/Wallet/index.tsx:
```diff
@@ -926,6 +938,61 @@ const Wallet = ({
+        },
+      ],
+      closeButtonOptions: {
+        label: 'x', // Hacky solution as there isn't a close icon button variant
```
Q: why not expliciteluy using the word "Close" ?

ameliejyc committed:
chore: add optin metrics test to boost coverage

ameliejyc committed:
Merge branch 'main' into ceux-730-pna25-messaging
